boefjes/boefjes/sql/db.py

```diff
@@ -16,15 +16,24 @@
 
 @cache
 def get_engine() -> Engine:
-    logger.info("Connecting to database..")
+    db_uri_redacted = _redact_db_uri(db_uri=str(settings.katalogus_db_uri))
+    logger.info("Connecting to database %s with pool size %s...", db_uri_redacted, settings.db_engine_pool_size)
 
-    engine = create_engine(make_url(str(settings.katalogus_db_uri)), pool_pre_ping=True, pool_size=25)
+    engine = create_engine(
+        make_url(str(settings.katalogus_db_uri)), pool_pre_ping=True, pool_size=settings.db_engine_pool_size
+    )
 
-    logger.info("Connected to database")
+    logger.info("Connected to database %s", db_uri_redacted)
 
     return engine
 
 
+def _redact_db_uri(db_uri: str, redact="<secret/>") -> str:
+    """Redacts db_uri strings of form $x:$y:secret@$z to $x:$y:$redact@$z."""
+    pre, post = db_uri.split("@")
+    return pre.rsplit(":", 1)[0] + ":" + redact + "@" + post
```

Comment:
Sqlalchemy already implements this. `make_url` returns a URL object and that has a `render_as_string` method dat hides the password by the default. That will also handle all the edge cases correctly, passwords aren't strictly required for example.

https://docs.sqlalchemy.org/en/20/core/engines.html#sqlalchemy.engine.URL.render_as_string